File: reporters/javascript/src/runtime-config.ts
const env = typeof process !== "undefined" ? process.env : {};

export const runtimeConfig = {
  enabled: env.TT_DISABLE !== "true",
  serverUrl: env.TT_SERVER_URL ?? "wss://tracethat.dev",
  token: env.TT_TOKEN ?? (null as string | null),
  verbose: env.TT_VERBOSE === "true",
  debug: env.TT_DEBUG === 'true',
};

if (runtimeConfig.debug) {
  console.warn('[tracethat.dev] Running in debug mode, the traffic will be decrypted on the server!')
}

export const disableDevtools = () => {
  runtimeConfig.enabled = false;
  onConfigChangeCb?.();
};

export const setServerUrl = (newUrl) => {
  runtimeConfig.serverUrl = newUrl;
  onConfigChangeCb?.();
};

let onConfigChangeCb: () => void | null;
export const onConfigChange = (cb: () => void) => {
  // set the callback after initialization
  setTimeout(() => {
    onConfigChangeCb = cb;
  }, 100);
};

export const registerToken = (token: string) => {
  runtimeConfig.token = token;
  onConfigChangeCb?.();
};

export const setVerbose = (verbose: boolean) => {
  runtimeConfig.verbose = verbose;
}

File: reporters/python/runtime_config.py
import os 

class Config:
    server_url: str
    token: str | None
    enabled: bool

    def __init__(self):
        self.server_url = os.getenv('TT_SERVER_URL') or 'ws://localhost:3000'
        self.token = os.getenv('TT_TOKEN') or None
        self.enabled = os.getenv('TT_DISABLE') != 'true'

config = Config()

def load() -> Config:
    return config

def registerToken(token: str) -> None:
    config.token = token

def setServerUrl(url: str) -> None:
    config.server_url = url

def disable() -> None:
    config.enabled = False